### ISSUE TITLE
[interface-name-prefix] Allow to set custom prefix

### ISSUE THREAD

**Issue by antoine-pous** [typescript-eslint/typescript-eslint#1145]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/interface-name-prefix": [2, 
      {
        "usePrefix": "always",
        "prefix": "CustomPrefix"
      }
    ]
  }
}
```

```TS
export interface CustomPrefixHelloWorld {

}
```

**Expected Result**
This rule would be awesome if it allow us natively to use the prefix we need instead of having only `I`. I know it's the recommended stuff, but sometimes a team works with same practices and other keyworks.

**Actual Result**
We can use only `I` :disappointed:
